### PR TITLE
feat: add initial buildpack

### PR DESCRIPTION
**Pull request by ctrox**
This buildpack complements the [`buildpack-static-confgen`](https://github.com/ninech/buildpack-static-confgen) buildpack, see the `README.md` on why this can't be in a single buildpack. This simply requires nginx if an `index.html` can be found in the workspace. The build phase is a noop.

### CODE REVIEW

File: detect.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	publicDir := ""
+	for _, path := range paths {
+		f, err := os.Stat(filepath.Join(workingDir, path, indexFile))
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				continue
+			}
+			return "", packit.Fail.WithMessage("failed to stat: %w", err)
+		}
+		if f.IsDir() {
```
Comment (thirdeyenick) [inline detect.go:53]: not sure, but if there is directory `index.html` in the root, but there is an additional file `public/index.html` then we would not find it, right?

Comment (ctrox) [reply detect.go:53]: yes, but should we? Would be quite a strange pattern to put your file in a directory named `index.html` :smile:

File: README.md [hunk 2]
```diff
@@ -1 +1,17 @@
 # buildpack-static-require
+
+This buildpack complements the
+[`buildpack-static-confgen`](https://github.com/ninech/buildpack-static-confgen)
+buildpack, see the `README.md` on why this can't be in a single buildpack.
```
Comment (thirdeyenick) [inline README.md:5]: I assume you mean the `README.md` in the other repo, because I am just reading a `README.md` file?

File: detect.go (hunk 1 above)
Comment (thirdeyenick) [reply detect.go:53]: Well it wouldn't be. There would be a valid file in `public/index.html`, but additionally there is a directory called `index.html` in the root (maybe a mistake?).

Comment (ctrox) [reply detect.go:53]: ah that's what you mean. I guess we could just continue here instead of erroring.

File: README.md (hunk 2 above)
Comment (ctrox) [reply README.md:5]: yep, is it not clear enough with the link just before that?

File: detect.go (hunk 1 above)
Comment (ctrox) [reply detect.go:53]: I changed it to continue here.

File: README.md (hunk 2 above)
Comment (thirdeyenick) [reply README.md:5]: Yeah it is fine.

**Review by thirdeyenick — APPROVED** [review]

**Pull request merged**